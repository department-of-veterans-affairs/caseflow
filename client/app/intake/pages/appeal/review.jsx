import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import RadioField from '../../../components/RadioField';
import DateSelector from '../../../components/DateSelector';
import { Redirect } from 'react-router-dom';
import SelectClaimant from '../../components/SelectClaimant';
import LegacyOptInApproved from '../../components/LegacyOptInApproved';
import { setDocketType } from '../../actions/appeal';
import {
  setVeteranIsNotClaimant,
  setClaimant,
  setPayeeCode,
  setLegacyOptInApproved
} from '../../actions/decisionReview';
import { setReceiptDate } from '../../actions/intake';
import { PAGE_PATHS, INTAKE_STATES, FORM_TYPES } from '../../constants';
import { getIntakeStatus } from '../../selectors';
import ErrorAlert from '../../components/ErrorAlert';

class Review extends React.PureComponent {
  render() {
    const {
      appealStatus,
      veteranName,
      receiptDate,
      receiptDateError,
      docketType,
      docketTypeError,
      legacyOptInApproved,
      legacyOptInApprovedError,
      reviewIntakeError,
      featureToggles
    } = this.props;

    switch (appealStatus) {
    case INTAKE_STATES.NONE:
      return <Redirect to={PAGE_PATHS.BEGIN} />;
    case INTAKE_STATES.COMPLETED:
      return <Redirect to={PAGE_PATHS.COMPLETED} />;
    default:
    }

    const legacyOptInEnabled = featureToggles.legacyOptInEnabled;

    const docketTypeRadioOptions = [
      { value: 'direct_review',
        displayText: 'Direct Review' },
      { value: 'evidence_submission',
        displayText: 'Evidence Submission' },
      { value: 'hearing',
        displayText: 'Hearing' }
    ];

    return <div>
      <h1>Review { veteranName }'s { FORM_TYPES.APPEAL.name }</h1>

      { reviewIntakeError && <ErrorAlert /> }

      <DateSelector
        name="receipt-date"
        label="What is the Receipt Date of this form?"
        value={receiptDate}
        onChange={this.props.setReceiptDate}
        errorMessage={receiptDateError}
        strongLabel
      />

      <RadioField
        name="docket-type"
        label="Which review option did the Veteran request?"
        strongLabel
        vertical
        options={docketTypeRadioOptions}
        onChange={this.props.setDocketType}
        errorMessage={docketTypeError}
        value={docketType}
      />

      <SelectClaimantConnected />

      { legacyOptInEnabled && <LegacyOptInApproved
        value={legacyOptInApproved === null ? null : legacyOptInApproved.toString()}
        onChange={this.props.setLegacyOptInApproved}
        errorMessage={legacyOptInApprovedError}
      /> }
    </div>;
  }
}

const SelectClaimantConnected = connect(
  ({ appeal, intake }) => ({
    isVeteranDeceased: intake.veteran.isDeceased,
    veteranIsNotClaimant: appeal.veteranIsNotClaimant,
    veteranIsNotClaimantError: appeal.veteranIsNotClaimantError,
    claimant: appeal.claimant,
    claimantError: appeal.claimantError,
    payeeCode: appeal.payeeCode,
    relationships: appeal.relationships,
    benefitType: appeal.benefitType,
    formType: intake.formType
  }),
  (dispatch) => bindActionCreators({
    setVeteranIsNotClaimant,
    setClaimant,
    setPayeeCode
  }, dispatch)
)(SelectClaimant);

export default connect(
  (state) => ({
    veteranName: state.intake.veteran.name,
    appealStatus: getIntakeStatus(state),
    receiptDate: state.appeal.receiptDate,
    receiptDateError: state.appeal.receiptDateError,
    docketType: state.appeal.docketType,
    docketTypeError: state.appeal.docketTypeError,
    legacyOptInApproved: state.appeal.legacyOptInApproved,
    legacyOptInApprovedError: state.appeal.legacyOptInApprovedError,
    reviewIntakeError: state.appeal.requestStatus.reviewIntakeError
  }),
  (dispatch) => bindActionCreators({
    setDocketType,
    setReceiptDate,
    setLegacyOptInApproved
  }, dispatch)
)(Review);
