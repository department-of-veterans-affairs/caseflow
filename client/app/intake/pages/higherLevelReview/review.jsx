import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Redirect } from 'react-router-dom';
import RadioField from '../../../components/RadioField';
import DateSelector from '../../../components/DateSelector';
import BenefitType from '../../components/BenefitType';
import LegacyOptInApproved from '../../components/LegacyOptInApproved';
import SelectClaimant from '../../components/SelectClaimant';
import { setInformalConference, setSameOffice } from '../../actions/higherLevelReview';
import {
  setBenefitType,
  setVeteranIsNotClaimant,
  setClaimant,
  setPayeeCode,
  setLegacyOptInApproved
} from '../../actions/decisionReview';
import { setReceiptDate } from '../../actions/intake';
import { PAGE_PATHS, INTAKE_STATES, BOOLEAN_RADIO_OPTIONS, FORM_TYPES, VBMS_BENEFIT_TYPES } from '../../constants';
import { getIntakeStatus } from '../../selectors';
import ErrorAlert from '../../components/ErrorAlert';
import PropTypes from 'prop-types';

class Review extends React.PureComponent {
  render() {
    const {
      higherLevelReviewStatus,
      veteranName,
      receiptDate,
      receiptDateError,
      benefitType,
      benefitTypeError,
      informalConference,
      informalConferenceError,
      sameOffice,
      sameOfficeError,
      legacyOptInApproved,
      legacyOptInApprovedError,
      reviewIntakeError,
      veteranValid,
      veteranInvalidFields
    } = this.props;

    switch (higherLevelReviewStatus) {
    case INTAKE_STATES.NONE:
      return <Redirect to={PAGE_PATHS.BEGIN} />;
    case INTAKE_STATES.COMPLETED:
      return <Redirect to={PAGE_PATHS.COMPLETED} />;
    default:
    }

    const showInvalidVeteranError = !veteranValid && VBMS_BENEFIT_TYPES.includes(benefitType);

    return <div>
      <h1>Review { veteranName }'s { FORM_TYPES.HIGHER_LEVEL_REVIEW.name }</h1>

      { reviewIntakeError && <ErrorAlert /> }
      { showInvalidVeteranError && <ErrorAlert errorCode="veteran_not_valid" errorData={veteranInvalidFields} /> }

      <BenefitType
        value={benefitType}
        onChange={this.props.setBenefitType}
        errorMessage={benefitTypeError}
      />

      <DateSelector
        name="receipt-date"
        label="What is the Receipt Date of this form?"
        value={receiptDate}
        onChange={this.props.setReceiptDate}
        errorMessage={receiptDateError}
        type="date"
        strongLabel
      />

      <RadioField
        name="informal-conference"
        label="Was an informal conference requested?"
        strongLabel
        vertical
        options={BOOLEAN_RADIO_OPTIONS}
        onChange={this.props.setInformalConference}
        errorMessage={informalConferenceError}
        value={informalConference === null ? null : informalConference.toString()}
      />

      <RadioField
        name="same-office"
        label="Was an interview by the same office requested?"
        strongLabel
        vertical
        options={BOOLEAN_RADIO_OPTIONS}
        onChange={this.props.setSameOffice}
        errorMessage={sameOfficeError}
        value={sameOffice === null ? null : sameOffice.toString()}
      />

      <SelectClaimantConnected />

      <LegacyOptInApproved
        value={legacyOptInApproved === null ? null : legacyOptInApproved.toString()}
        onChange={this.props.setLegacyOptInApproved}
        errorMessage={legacyOptInApprovedError}
      />
    </div>;
  }
}

Review.propTypes = {
  veteranName: PropTypes.string,
  receiptDate: PropTypes.string,
  receiptDateError: PropTypes.string,
  benefitType: PropTypes.string,
  benefitTypeError: PropTypes.string,
  informalConference: PropTypes.string,
  informalConferenceError: PropTypes.string,
  sameOffice: PropTypes.string,
  sameOfficeError: PropTypes.string,
  legacyOptInApproved: PropTypes.bool,
  legacyOptInApprovedError: PropTypes.string,
  reviewIntakeError: PropTypes.string,
  veteranValid: PropTypes.bool,
  veteranInvalidFields: PropTypes.object,
  setBenefitType: PropTypes.func,
  setReceiptDate: PropTypes.func,
  setInformalConference: PropTypes.func,
  setSameOffice: PropTypes.func,
  setLegacyOptInApproved: PropTypes.func,
  higherLevelReviewStatus: PropTypes.string
};

const SelectClaimantConnected = connect(
  ({ higherLevelReview, intake }) => ({
    isVeteranDeceased: intake.veteran.isDeceased,
    veteranIsNotClaimant: higherLevelReview.veteranIsNotClaimant,
    veteranIsNotClaimantError: higherLevelReview.veteranIsNotClaimantError,
    claimant: higherLevelReview.claimant,
    claimantError: higherLevelReview.claimantError,
    payeeCode: higherLevelReview.payeeCode,
    payeeCodeError: higherLevelReview.payeeCodeError,
    relationships: higherLevelReview.relationships,
    benefitType: higherLevelReview.benefitType,
    formType: intake.formType
  }),
  (dispatch) => bindActionCreators({
    setVeteranIsNotClaimant,
    setClaimant,
    setPayeeCode
  }, dispatch)
)(SelectClaimant);

export default connect(
  (state) => ({
    veteranName: state.intake.veteran.name,
    higherLevelReviewStatus: getIntakeStatus(state),
    receiptDate: state.higherLevelReview.receiptDate,
    receiptDateError: state.higherLevelReview.receiptDateError,
    benefitType: state.higherLevelReview.benefitType,
    benefitTypeError: state.higherLevelReview.benefitTypeError,
    legacyOptInApproved: state.higherLevelReview.legacyOptInApproved,
    legacyOptInApprovedError: state.higherLevelReview.legacyOptInApprovedError,
    informalConference: state.higherLevelReview.informalConference,
    informalConferenceError: state.higherLevelReview.informalConferenceError,
    sameOffice: state.higherLevelReview.sameOffice,
    sameOfficeError: state.higherLevelReview.sameOfficeError,
    reviewIntakeError: state.higherLevelReview.requestStatus.reviewIntakeError,
    veteranValid: state.higherLevelReview.veteranValid,
    veteranInvalidFields: state.higherLevelReview.veteranInvalidFields
  }),
  (dispatch) => bindActionCreators({
    setInformalConference,
    setSameOffice,
    setReceiptDate,
    setBenefitType,
    setLegacyOptInApproved
  }, dispatch)
)(Review);
