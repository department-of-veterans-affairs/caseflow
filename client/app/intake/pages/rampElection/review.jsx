import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import RadioField from '../../../components/RadioField';
import DateSelector from '../../../components/DateSelector';
import { Redirect } from 'react-router-dom';
import _ from 'lodash';
import { setReceiptDate, setOptionSelected } from '../../actions/intake';
import { PAGE_PATHS, INTAKE_STATES, REVIEW_OPTIONS } from '../../constants';
import { getIntakeStatus } from '../../selectors';
import ErrorAlert from '../../components/ErrorAlert';
import PropTypes from 'prop-types';

class Review extends React.PureComponent {
  render() {
    const {
      rampElectionStatus,
      veteranName,
      optionSelected,
      optionSelectedError,
      receiptDate,
      receiptDateError,
      reviewIntakeError
    } = this.props;

    switch (rampElectionStatus) {
    case INTAKE_STATES.NONE:
      return <Redirect to={PAGE_PATHS.BEGIN} />;
    case INTAKE_STATES.COMPLETED:
      return <Redirect to={PAGE_PATHS.COMPLETED} />;
    default:
    }

    const rampElectionReviewOptions = _.reject(REVIEW_OPTIONS, REVIEW_OPTIONS.APPEAL);
    const radioOptions = _.map(rampElectionReviewOptions, (option) => ({
      value: option.key,
      displayText: option.name
    }));

    return <div>
      <h1>Review { veteranName }'s Opt-In Election Form</h1>

      { reviewIntakeError && <ErrorAlert /> }

      <DateSelector
        name="receipt-date"
        label="What is the Receipt Date of this form?"
        value={receiptDate}
        onChange={this.props.setReceiptDate}
        errorMessage={receiptDateError}
        type="date"
        strongLabel
      />

      <RadioField
        name="opt-in-election"
        label="Which review lane did the Veteran select?"
        strongLabel
        options={radioOptions}
        onChange={this.props.setOptionSelected}
        errorMessage={optionSelectedError}
        value={optionSelected}
      />
    </div>;
  }
}

Review.propTypes = {
  veteranName: PropTypes.string,
  receiptDate: PropTypes.string,
  receiptDateError: PropTypes.string,
  optionSelected: PropTypes.string,
  optionSelectedError: PropTypes.string,
  setReceiptDate: PropTypes.func,
  setOptionSelected: PropTypes.func,
  rampElectionStatus: PropTypes.string,
  reviewIntakeError: PropTypes.string
};

export default connect(
  (state) => ({
    veteranName: state.intake.veteran.name,
    rampElectionStatus: getIntakeStatus(state),
    optionSelected: state.rampElection.optionSelected,
    optionSelectedError: state.rampElection.optionSelectedError,
    receiptDate: state.rampElection.receiptDate,
    receiptDateError: state.rampElection.receiptDateError,
    reviewIntakeError: state.rampElection.requestStatus.reviewIntakeError
  }),
  (dispatch) => bindActionCreators({
    setOptionSelected,
    setReceiptDate
  }, dispatch)
)(Review);
