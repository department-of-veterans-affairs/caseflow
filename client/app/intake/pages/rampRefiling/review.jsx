import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import CancelButton from '../../components/CancelButton';
import RadioField from '../../../components/RadioField';
import DateSelector from '../../../components/DateSelector';
import Button from '../../../components/Button';
import Alert from '../../../components/Alert';
import { Redirect } from 'react-router-dom';
import _ from 'lodash';
import { REQUEST_STATE, PAGE_PATHS, INTAKE_STATES, REVIEW_OPTIONS } from '../../constants';
import { setAppealDocket, submitReview, confirmIneligibleForm } from '../../actions/rampRefiling';
import { setReceiptDate, setOptionSelected } from '../../actions/common';
import { toggleIneligibleError } from '../../util';
import { getIntakeStatus } from '../../selectors';

class Review extends React.PureComponent {
  beginNextIntake = () => {
    this.props.confirmIneligibleForm(this.props.intakeId);
  }
  render() {
    const {
      rampRefilingStatus,
      veteranName,
      optionSelected,
      optionSelectedError,
      hasInvalidOption,
      receiptDate,
      receiptDateError,
      appealDocket,
      appealDocketError,
      submitInvalidOptionError
    } = this.props;

    switch (rampRefilingStatus) {
    case INTAKE_STATES.NONE:
      return <Redirect to={PAGE_PATHS.BEGIN} />;
    case INTAKE_STATES.COMPLETED:
      return <Redirect to={PAGE_PATHS.COMPLETED} />;
    default:
    }

    const reviewRadioOptions = _.map(REVIEW_OPTIONS, (option) => ({
      value: option.key,
      displayText: option.name
    }));

    const docketRadioOptions = [
      {
        value: 'direct_review',
        displayText: 'Direct Review'
      },
      {
        value: 'evidence_submission',
        displayText: 'Evidence Submission'
      },
      {
        value: 'hearing',
        displayText: 'Hearing'
      }
    ];

    return <div>
      { submitInvalidOptionError && <Alert title="Something went wrong" type="error">
        Please try again. If the problem persists, please contact Caseflow support.</Alert>}

      { toggleIneligibleError(hasInvalidOption, optionSelected) &&
        <Alert title="Ineligible for Higher-Level Review" type="error" lowerMargin>
          Contact the Veteran to verify their lane selection. If you are unable to reach
          the Veteran, send a letter indicating that their selected lane is not available,
          and that they may clarify their lane selection within 30 days. <br />
          <Button
            name="begin-next-intake"
            onClick={this.beginNextIntake}
            loading={this.props.requestState === REQUEST_STATE.IN_PROGRESS}
            legacyStyling={false}>
            Begin next intake
          </Button>
        </Alert>
      }

      <h1>Review { veteranName }'s 21-4138 RAMP Selection Form</h1>

      <DateSelector
        name="receipt-date"
        label="What is the Receipt Date of this form?"
        value={receiptDate}
        onChange={this.props.setReceiptDate}
        errorMessage={receiptDateError}
        strongLabel
      />

      <RadioField
        name="opt-in-election"
        label="Which review lane did the Veteran select?"
        strongLabel
        options={reviewRadioOptions}
        onChange={this.props.setOptionSelected}
        errorMessage={optionSelectedError}
        value={optionSelected}
      />

      { optionSelected === REVIEW_OPTIONS.APPEAL.key &&
        <RadioField
          name="appeal-docket"
          label="Which type of appeal did the Veteran request?"
          strongLabel
          options={docketRadioOptions}
          onChange={this.props.setAppealDocket}
          errorMessage={appealDocketError}
          value={appealDocket}
        />
      }
    </div>;
  }
}

export default connect(
  (state) => ({
    veteranName: state.intake.veteran.name,
    rampRefilingStatus: getIntakeStatus(state),
    optionSelected: state.rampRefiling.optionSelected,
    optionSelectedError: state.rampRefiling.optionSelectedError,
    hasInvalidOption: state.rampRefiling.hasInvalidOption,
    receiptDate: state.rampRefiling.receiptDate,
    receiptDateError: state.rampRefiling.receiptDateError,
    requestStatus: state.rampRefiling.requestStatus,
    intakeId: state.intake.id,
    appealDocket: state.rampRefiling.appealDocket,
    appealDocketError: state.rampRefiling.appealDocketError,
    submitInvalidOptionError: state.rampRefiling.submitInvalidOptionError
  }),
  (dispatch) => bindActionCreators({
    setOptionSelected,
    setReceiptDate,
    setAppealDocket,
    confirmIneligibleForm
  }, dispatch)
)(Review);

class ReviewNextButton extends React.PureComponent {
  handleClick = () => {
    this.props.submitReview(this.props.intakeId, this.props.rampRefiling).then(
      () => {
        this.props.history.push('/finish');
      }
    ).
      catch((error) => error);
  }

  render = () =>
    <Button
      name="submit-review"
      onClick={this.handleClick}
      loading={this.props.requestState === REQUEST_STATE.IN_PROGRESS}
      legacyStyling={false}
      disabled={toggleIneligibleError(this.props.hasInvalidOption, this.props.optionSelected)}
    >
      Continue to next step
    </Button>;
}

const ReviewNextButtonConnected = connect(
  ({ rampRefiling, intake }) => ({
    intakeId: intake.id,
    requestState: rampRefiling.requestStatus.submitReview,
    rampRefiling,
    hasInvalidOption: rampRefiling.hasInvalidOption,
    optionSelected: rampRefiling.optionSelected
  }),
  (dispatch) => bindActionCreators({
    submitReview
  }, dispatch)
)(ReviewNextButton);

export class ReviewButtons extends React.PureComponent {
  render = () =>
    <div>
      <CancelButton />
      <ReviewNextButtonConnected history={this.props.history} />
    </div>
}
