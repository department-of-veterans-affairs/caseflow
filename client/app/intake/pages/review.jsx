import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import RadioField from '../../components/RadioField';
import DateSelector from '../../components/DateSelector';
import Button from '../../components/Button';
import { setSelectedOption, setReceiptDate, submitReview } from '../redux/actions';
import { REQUEST_STATE } from '../constants';

class Review extends React.PureComponent {
  render() {
    const radioOptions = [
      {
        value: 'supplemental_claim',
        displayText: 'Supplemental Claim'
      },
      {
        value: 'higher_level_review_with_hearing',
        displayElem: <span>Higher Level Review <strong>with</strong> DRO hearing request</span>
      },
      {
        value: 'higher_level_review',
        displayElem: <span>Higher Level Review with<strong>out</strong> DRO hearing request</span>
      },
      {
        value: 'withdraw',
        displayText: 'Withdraw all pending appeals'
      }
    ];

    return <div>
      <h1>Review { this.props.veteran.name }'s opt-in request</h1>
      <p>Check the Veteran's RAMP Opt-In Election form in the Centralized Portal.</p>

      <RadioField
        name="opt-in-election"
        label="Which election did the Veteran select?"
        strongLabel
        options={radioOptions}
        onChange={this.props.setSelectedOption}
        value={this.props.rampElection.optionSelected}
      />

      <DateSelector
        name="receipt-date"
        label="What is the Receipt Date for this election form?"
        value={this.props.rampElection.receiptDate}
        onChange={this.props.setReceiptDate}
        strongLabel
      />
    </div>;
  }
}

class ReviewNextButton extends React.PureComponent {
  handleClick = () => {
    this.props.submitReview(this.props.rampElection).then(
      () => this.props.history.push('/finish')
    );
  }

  render = () =>
    <Button
      name="submit-review"
      onClick={this.handleClick}
      loading={this.props.requestState === REQUEST_STATE.IN_PROGRESS}
    >
      Continue to next step
    </Button>;
}

const ReviewNextButtonConnected = connect(
  ({ rampElection, requestStatus }) => ({
    requestState: requestStatus.submitReview,
    rampElection
  }),
  (dispatch) => bindActionCreators({
    submitReview
  }, dispatch)
)(ReviewNextButton);

export { ReviewNextButtonConnected as ReviewNextButton };

export default connect(
  ({ veteran, rampElection }) => ({
    veteran,
    rampElection
  }),
  (dispatch) => bindActionCreators({
    setSelectedOption,
    setReceiptDate
  }, dispatch)
)(Review);
