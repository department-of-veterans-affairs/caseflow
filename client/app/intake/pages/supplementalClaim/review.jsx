import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import DateSelector from '../../../components/DateSelector';
import CancelButton from '../../components/CancelButton';
import { Redirect } from 'react-router-dom';
import Button from '../../../components/Button';
import { submitReview } from '../../actions/supplementalClaim';
import { setReceiptDate } from '../../actions/common';
import { REQUEST_STATE, PAGE_PATHS, INTAKE_STATES } from '../../constants';
import { getIntakeStatus } from '../../selectors';

class Review extends React.PureComponent {
  render() {
    const {
      supplementalClaimStatus,
      veteranName,
      receiptDate,
      receiptDateError
    } = this.props;

    switch (supplementalClaimStatus) {
    case INTAKE_STATES.NONE:
      return <Redirect to={PAGE_PATHS.BEGIN} />;
    case INTAKE_STATES.COMPLETED:
      return <Redirect to={PAGE_PATHS.COMPLETED} />;
    default:
    }

    return <div>
      <h1>Review { veteranName }'s Supplemental Claim (VA Form 21-526b)</h1>

      <DateSelector
        name="receipt-date"
        label="What is the Receipt Date of this form?"
        value={receiptDate}
        onChange={this.props.setReceiptDate}
        errorMessage={receiptDateError}
        strongLabel
      />

    </div>;
  }
}

class ReviewNextButton extends React.PureComponent {
  handleClick = () => {
    this.props.submitReview(this.props.intakeId, this.props.supplementalClaim).then(
      () => this.props.history.push('/finish')
    );
  }

  render = () =>
    <Button
      name="submit-review"
      onClick={this.handleClick}
      loading={this.props.requestState === REQUEST_STATE.IN_PROGRESS}
      legacyStyling={false}
    >
      Continue to next step
    </Button>;
}

const ReviewNextButtonConnected = connect(
  ({ supplementalClaim, intake }) => ({
    intakeId: intake.id,
    requestState: supplementalClaim.requestStatus.submitReview,
    supplementalClaim
  }),
  (dispatch) => bindActionCreators({
    submitReview
  }, dispatch)
)(ReviewNextButton);

export class ReviewButtons extends React.PureComponent {
  render = () =>
    <div>
      <CancelButton />
      <ReviewNextButtonConnected history={this.props.history} />
    </div>
}

export default connect(
  (state) => ({
    veteranName: state.intake.veteran.name,
    supplementalClaimStatus: getIntakeStatus(state),
    receiptDate: state.supplementalClaim.receiptDate,
    receiptDateError: state.supplementalClaim.receiptDateError
  }),
  (dispatch) => bindActionCreators({
    setReceiptDate
  }, dispatch)
)(Review);
