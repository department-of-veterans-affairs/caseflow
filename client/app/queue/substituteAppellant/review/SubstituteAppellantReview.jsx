import React from 'react';
import PropTypes from 'prop-types';

import AppSegment from '@department-of-veterans-affairs/caseflow-frontend-toolkit/components/AppSegment';
import {
  SUBSTITUTE_APPELLANT_REVIEW_TITLE,
  SUBSTITUTE_APPELLANT_REVIEW_SUBHEAD,
} from 'app/../COPY';
import CheckoutButtons from 'app/queue/docketSwitch/grant/CheckoutButtons';
import { pageHeader } from '../styles';
import { format, parseISO } from 'date-fns';
import { css } from 'glamor';
import { formatDateStr } from '../../../util/DateUtil';

const styles = {
  mainTable: css({
    '& .bolded-header': {
      fontWeight: 'bold',
    },
    '& tr > td': {
      width: '50%',
    },
    '&': {
      margin: 0,
    },
    '& tr:first-of-type td': {
      borderTop: '1px solid #D6D7D9',
    },
    '& tr:last-of-type td': {
      paddingBottom: '20px',
    },
  }),
  tableSection: css({
    marginBottom: '40px'
  })
};

export const SubstituteAppellantReview = ({
  selectedTasks,
  existingValues,
  evidenceSubmissionEndDate,
  relationship,
  onBack,
  onCancel,
  onSubmit,
}) => {
  const substitutionDate = format(
    parseISO(existingValues.substitutionDate),
    'MM/dd/yyyy'
  );

  return (
    <>
      <AppSegment filledBackground>
        <section className={pageHeader}>
          <h1>{SUBSTITUTE_APPELLANT_REVIEW_TITLE}</h1>
          <div>{SUBSTITUTE_APPELLANT_REVIEW_SUBHEAD}</div>
        </section>
        <section className={styles.tableSection}>
          <h2>About the appellant</h2>
          <table className={`usa-table-borderless ${styles.mainTable}`}>
            <tbody>
              <tr>
                <td className="bolded-header">
                  Substitution granted by the RO
                </td>
                <td>{existingValues.substitutionDate && substitutionDate}</td>
              </tr>
              <tr>
                <td className="bolded-header">Name</td>
                <td>{relationship && relationship.fullName}</td>
              </tr>
              <tr>
                <td className="bolded-header">Relation to Veteran</td>
                <td>{relationship && relationship.relationshipType}</td>
              </tr>
            </tbody>
          </table>
        </section>
        <section className={styles.tableSection}>
          <h2>Reactivated tasks</h2>
          {selectedTasks.length > 0 && (
            <table className={`usa-table-borderless ${styles.mainTable}`}>
              <tbody>
                {selectedTasks.map((task) => {
                  return (
                    <tr className="task-detail" key={`${task.taskId}`}>
                      <td>
                        {task.label.replace('Task', '')}
                      </td>
                      <td>
                        {task.type === 'EvidenceSubmissionWindowTask' && (
                          <span className="bolded-header">End date: </span>
                        )}
                        {task.type === 'EvidenceSubmissionWindowTask' &&
                          formatDateStr(evidenceSubmissionEndDate, 'YYYY-MM-DD', 'MM/DD/YYYY')}
                      </td>
                    </tr>
                  );
                })}
              </tbody>
            </table>
          )}
        </section>
      </AppSegment>
      <div className="controls cf-app-segment">
        <CheckoutButtons
          onCancel={onCancel}
          onBack={onBack}
          onSubmit={onSubmit}
          submitText="Confirm"
        />
      </div>
    </>
  );
};
SubstituteAppellantReview.propTypes = {
  selectedTasks: PropTypes.arrayOf(
    PropTypes.shape({
      taskId: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
      label: PropTypes.string,
    })
  ),
  existingValues: PropTypes.shape({
    substitutionDate: PropTypes.string,
    participantId: PropTypes.string,
    taskIds: PropTypes.array,
  }),
  evidenceSubmissionEndDate: PropTypes.string,
  relationship: PropTypes.shape({
    fullName: PropTypes.string,
    relationshipType: PropTypes.string,
  }),
  onBack: PropTypes.func,
  onCancel: PropTypes.func,
  onSubmit: PropTypes.func,
};
