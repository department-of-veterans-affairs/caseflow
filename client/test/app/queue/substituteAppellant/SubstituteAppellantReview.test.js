import { axe } from 'jest-axe';
import React from 'react';

import { SubstituteAppellantReview } from 'app/queue/substituteAppellant/review/SubstituteAppellantReview';
import { queueWrapper as Wrapper } from '../../../../test/data/stores/queueStore';
import { MemoryRouter } from 'react-router';
import { render, screen } from '@testing-library/react';
import { sampleTasksForDismissedEvidenceSubmissionDocket } from 'test/data/queue/substituteAppellant/tasks';
import userEvent from '@testing-library/user-event';

describe('SubstituteAppellantReview', () => {
  const onBack = jest.fn();
  const onCancel = jest.fn();
  const onSubmit = jest.fn();
  const sampleTasks = sampleTasksForDismissedEvidenceSubmissionDocket();
  const selectedTaskIds = sampleTasks.slice(2).map((task) => task.taskId);

  const defaults = {
    selectedTasks: sampleTasks,
    existingValues: { substitutionDate: '2021-03-25',
      participantId: 'particip-id',
      taskIds: selectedTaskIds },
    evidenceSubmissionEndDate: new Date('2021-05-30'),
    onBack,
    onCancel,
    onSubmit
  };

  const storeArgs = {
    substituteAppellant: {
      relationships: [
        { value: '"1129318238"',
          fullName: 'Cathy Smith',
          relationshipType: 'Child',
          displayText: 'Cathy Smith, Child',
        },
      ],
    },
  };

  const setup = (props) =>
    render(
      <MemoryRouter>
        <Wrapper {...storeArgs}>
          <SubstituteAppellantReview {...defaults} {...props} />
        </Wrapper>
      </MemoryRouter>
    );

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });

  it('fires onCancel', async () => {
    setup();
    expect(onCancel).not.toHaveBeenCalled();

    await userEvent.click(screen.getByRole('button', { name: /cancel/i }));
    expect(onCancel).toHaveBeenCalled();
  });

  it('passes a11y testing', async () => {
    const { container } = setup();

    const results = await axe(container);

    expect(results).toHaveNoViolations();
  });
});
