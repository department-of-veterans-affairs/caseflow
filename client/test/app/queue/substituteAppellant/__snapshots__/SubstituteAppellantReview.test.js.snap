// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubstituteAppellantReview renders correctly 1`] = `
<div>
  <section
    class="cf-app-segment cf-app-segment--alt"
  >
    <section
      class="css-1kmvzpy"
    >
      <h1>
        Review and confirm
      </h1>
      <div>
        Review the details below to make sure the information is correct before submitting. If you need to make changes, please go back to the associated section.
      </div>
    </section>
    <section
      class="css-1urdhzr"
    >
      <h2>
        About the appellant
      </h2>
      <table
        class="usa-table-borderless css-x3zbi1"
      >
        <tbody>
          <tr>
            <td
              class="bolded-header"
            >
              Substitution granted by the RO
            </td>
            <td>
              03/25/2021
            </td>
          </tr>
          <tr>
            <td
              class="bolded-header"
            >
              Name
            </td>
            <td />
          </tr>
          <tr>
            <td
              class="bolded-header"
            >
              Relation to Veteran
            </td>
            <td />
          </tr>
        </tbody>
      </table>
    </section>
    <section
      class="css-1urdhzr"
    >
      <h2>
        Reactivated tasks
      </h2>
      <table
        class="usa-table-borderless css-x3zbi1"
      >
        <tbody>
          <tr
            class="task-detail"
          >
            <td>
              Root 
            </td>
            <td />
          </tr>
          <tr
            class="task-detail"
          >
            <td>
              Distribution 
            </td>
            <td />
          </tr>
          <tr
            class="task-detail"
          >
            <td>
              Evidence Submission Window 
            </td>
            <td>
              <span
                class="bolded-header"
              >
                End date: 
              </span>
              05/29/2021
            </td>
          </tr>
          <tr
            class="task-detail"
          >
            <td>
              Assign
            </td>
            <td />
          </tr>
          <tr
            class="task-detail"
          >
            <td>
              Review
            </td>
            <td />
          </tr>
          <tr
            class="task-detail"
          >
            <td>
              Draft decision
            </td>
            <td />
          </tr>
          <tr
            class="task-detail"
          >
            <td>
              Board Dispatch
            </td>
            <td />
          </tr>
          <tr
            class="task-detail"
          >
            <td>
              Board Dispatch
            </td>
            <td />
          </tr>
        </tbody>
      </table>
    </section>
  </section>
  <div
    class="controls cf-app-segment"
  >
    <div
      data-css-63oe3q=""
    >
      <span>
        <button
          class="cf-right-side usa-button"
          data-css-wn5k45=""
          id="button-next-button"
          type="button"
        >
          Confirm
        </button>
      </span>
      <button
        class="cf-btn-link usa-button"
        id="button-cancel-button"
        type="button"
      >
        Cancel
      </button>
      <button
        class="cf-right-side cf-prev-step usa-button-secondary usa-button"
        id="button-back-button"
        type="button"
      >
        Back
      </button>
    </div>
  </div>
</div>
`;
